### PR TITLE
Feat: Update to new architecture

### PR DESCRIPTION
**Pull request by heidipuk**
Update the structure of the template to follow the one proposed by the Monstar Lab Vapor team.
- The updated structure proposes a monolithic structure with targets for each domain of your API.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -11,35 +20,85 @@ let package = Package(
...
             name: "App",
             dependencies: [
+                "Auth",
+                "Core",
+                "User",
                 .product(name: "Bugsnag", package: "bugsnag"),
-                .product(name: "Fluent", package: "fluent"),
-                .product(name: "FluentPostgresDriver", package: "fluent-postgres-driver"),
+                .product(name: "Fluent", package: "fluent"), // remove
+                .product(name: "FluentPostgresDriver", package: "fluent-postgres-driver"), // remove
```
Comment (emrahsifoglu) [inline Package.swift]: Will they be removed in another pr?

File: Sources/App/Configuration/Environment.swift [hunk 2]
```diff
@@ -1,57 +1,78 @@
...
+            }
 
-    static var appUserRefreshTokenExpiration: TimeInterval {
-        self.get("APP_USER_REFRESH_TOKEN_EXPIRATION").flatMap { TimeInterval($0) } ?? 3600 * 24 * 365
-    }
-}
+            static var expiration: TimeInterval {
+                TimeInterval(assertGet("APP_USER_ACCESS_TOKEN_EXPIRATION")) ?? 3600
+            }
+        }
```
Comment (emrahsifoglu) [inline Sources/App/Configuration/Environment.swift:18]: There seems to issue with indentation.

File: Package.swift (hunk 1 above)
Comment (heidipuk) [reply Package.swift]: good catch! completely forgot to take those out 😅

File: Sources/App/Configuration/Environment.swift (hunk 2 above)
Comment (heidipuk) [reply Sources/App/Configuration/Environment.swift:18]: the indentations here are correct, the env vars have been re-organized a bit so they are now indented one level deeper than before

File: Dockerfile [hunk 3]
```diff
@@ -37,7 +37,7 @@ RUN cp "$(swift build --package-path /build -c release --show-bin-path)/Run" ./
 # ================================
 # Run image
 # ================================
-FROM swift:5.3-focal-slim
+FROM swift:5.4-focal-slim
```
Comment (heidipuk) [inline Dockerfile]:
```suggestion
FROM swift:5.5-focal-slim
```

File: Sources/App/Middleware/ErrorMiddleware+customize.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                let field: String
+                let message: String
+            }
+            let message: String
+            let validationErrors: [ValidationError]?
+        }
+        let error: ErrorPayload
+    }
+
+    public static func customize(environment: Environment) -> ErrorMiddleware {
```
Comment (siemensikkema) [inline Sources/App/Middleware/ErrorMiddleware+customize.swift:16]: What's the reasoning behind the name `customize`?

File: Sources/App/Configuration/NMeta+configure.swift [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import NMeta
+import Vapor
+
+extension Application.NMeta {
+    mutating func configure() {
+        exceptPaths = ["/health",]
```
Comment (siemensikkema) [inline Sources/App/Configuration/NMeta+configure.swift:6]:
```suggestion
        exceptPaths = [
            "/health",
        ]
```

File: Sources/App/Configuration/Environment.swift [hunk 6]
```diff
@@ -1,57 +1,78 @@
...
 
-// MARK: App User (Re)set Token
-extension Environment {
-    static var appUserResetTokenSignerKey: Data {
-        assertGet("APP_USER_RESET_TOKEN_SIGNER_KEY").data(using: .utf8)!
+        // MARK: App User Refresh Token
+
+        struct RefreshToken {
+            static var signerKey: Data {
```
Comment (siemensikkema) [inline Sources/App/Configuration/Environment.swift:23]: Let's alphabetize the static vars and structs in this file to set the right example 🙂

**Review by siemensikkema — COMMENTED** [review]
Thanks for taking this on! Besides some minor inline comments I noted the following down:

- Update License
    - Nodes Agency -> Monstarlab
    - 2020 -> 2021 since we're changing the copyright holder name
- update manifest.yml
    - name
    - update file structure to include when doing `vapor new`
- update README.md
    - name
